Shared/Windows/WinBtStream.cs

```diff
@@ -289,13 +292,71 @@ public static List<DeviceInfo> GetPaths()
                             //    AssociatedStack.Add(diDetail.devicePath, associatedStack);
                             //}
 
-                            result.Add(new DeviceInfo
+                            var nintrollerType = attrib.ProductID == 0x0330 ? ControllerType.ProController : ControllerType.Wiimote;
+                            var deviceName = String.Empty;
+                            var serialNumber = String.Empty;
+                            var buffer = new byte[128 * sizeof(char)];                                
+                            if (HidD_GetProductString(hidHandle, buffer, (uint)buffer.Length))
+                            {
+                                deviceName = System.Text.UnicodeEncoding.Unicode.GetString(buffer).TrimEnd('\0');
+                            }
+
+                            buffer = new byte[128 * sizeof(char)];
+                            if (HidD_GetSerialNumberString(hidHandle, buffer, (uint)buffer.Length))
+                            {
+                                serialNumber = System.Text.UnicodeEncoding.Unicode.GetString(buffer).TrimEnd('\0');
+                            }
+
+                            var newDeviceInfo = new DeviceInfo
                             {
                                 DevicePath = diDetail.devicePath,
-                                Type = attrib.ProductID == 0x0330 ? ControllerType.ProController : ControllerType.Wiimote,
+                                DeviceName = deviceName,
+                                SerialNumber = serialNumber,
+                                Type = nintrollerType,
                                 VID = attrib.VendorID.ToString("X4"),
                                 PID = attrib.ProductID.ToString("X4")
-                            });
+                            };
+
+                            // note: newer Wiimotes may identify as ProControllers via their ProductID
+                            // they reveal to be Wiimote only in their extension/status report
+                            // so make one roundtrip to ask the controller for its actual type
+                            using (var _stream = new WinBtStream(newDeviceInfo.DevicePath))

```

Comment:
There is already a measure in place that will make the icon be whatever the last controller type was detected for the same controller so they really just have to do a first time connect. If we're unsure about the controller type would could add a "?" badge on the icon to indicate it may or may not be the controller indicated.